contracts/proxy-lib/tests/common/proxy_lib_helper.rs

```diff
@@ -40,35 +40,35 @@ impl ProxyContractHelper {
             .await
     }
 
+    pub fn create_proposal(
+        &self,
+        author: &SigningKey,
+        receiver: &Account,
+        actions: Vec<ProposalAction>,
+    ) -> Result<Signed<Proposal>> {
+        let proposal = Proposal {
+            receiver_id: receiver.id().clone(),
+            author_id: author.verifying_key().to_bytes().rt().expect("Invalid signer"),
+            actions
+        };
+        let signed = Signed::new(&{proposal}, |p| author.sign(p))?;
+        Ok(signed)
+    }
+
     pub async fn create_and_approve_proposal(
         &self,
         caller: &Account,
-        signer: &SigningKey,
-        proposal: &Proposal,
-    ) -> Result<ExecutionFinalResult, near_workspaces::error::Error> {
-        let signer_id = signer
-            .verifying_key()
-            .to_bytes()
-            .rt()
-            .expect("Invalid signer");
-        let args = Signed::new(
-            &{
-                ProposalWithSigner {
-                    signer_id,
-                    proposal: proposal.clone(),
-                }
-            },
-            |p| signer.sign(p),
-        )
-        .expect("Failed to sign proposal");
-        caller
+        proposal: &Signed<Proposal>,
+    ) -> Result<ExecutionFinalResult> {
+        let call = caller
             .call(self.proxy_contract.id(), "create_and_approve_proposal")
             .args_json(json!({
-                "proposal": args
+                "proposal": proposal
             }))
             .max_gas()
             .transact()
-            .await
+            .await;
+        Ok(call?)
```

Comment:
interesting model 😅 

typically we do something like:

```suggestion
            .await;
        call.map_err(Into::into)
```

but here you can just bubble it up at the point you got the result, so it's clearer

```suggestion
            .await?;
        Ok(call)
```

Reply:
:D